src/crypto/browser.ts

```diff
@@ -0,0 +1,122 @@
+import * as ed25519 from 'noble-ed25519'
+import rsaOperations from 'keystore-idb/rsa'
+import utils from 'keystore-idb/utils'
+import aes from 'keystore-idb/aes'
+import { CharSize, SymmKeyLength } from 'keystore-idb/types'
+
+import { assertBrowser } from '../common/browser'
+import * as keystore from '../keystore'
+
+
+export const encrypt = async (data: Uint8Array, keyStr: string): Promise<Uint8Array> => {
+  assertBrowser('aes.encrypt')
+  const key = await aes.importKey(keyStr, { length: SymmKeyLength.B256 })
+  const encrypted = await aes.encryptBytes(data.buffer, key)
+  return new Uint8Array(encrypted)
+}
+
+export const decrypt = async (encrypted: Uint8Array, keyStr: string): Promise<Uint8Array> => {
+  assertBrowser('aes.decrypt')
+  const key = await aes.importKey(keyStr, { length: SymmKeyLength.B256 })
+  const decryptedBuf = await aes.decryptBytes(encrypted.buffer, key)
+  return new Uint8Array(decryptedBuf)
+}
+
+export const genKeyStr = async (): Promise<string> => {
+  assertBrowser('aes.genKeyStr')
+  const key = await aes.makeKey({ length: SymmKeyLength.B256 })
+  return aes.exportKey(key)
+}
+
+export const decryptGCM = async (encrypted: string, keyStr: string, ivStr: string): Promise<string> => {
```

Comment:
This feels a bit clunky and I'm not a huge fan. But we're using `AES-CTR` in the FS for now and `AES-GCM` for the auth lobby. Once we had an algo description to links in the FS, then we can standardize on `AES-GCM` for backwards-compatibility.

Also: in `keystore-idb`, I just prepend the `iv` to the encrypted message. So there is a slightly different api between these two